abel/three_point.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+from itertools import product
+
+def iabel_three_point_transform(IM):
+    """ Inverse Abel transformation using the algorithm of: 
+        Dasch, Applied Optics, Vol. 31, No. 8, 1146-1152 (1992).
+
+        Parameters:
+        ----------
+         - IM: a rows x cols numpy array 
+
+         Returns:
+         --------
+         - inv_IM: a rows x cols numpy array containing the Abel inversion of IM
+    """
+    IM = np.atleast_2d(IM)
+    row, col = IM.shape
+    D = np.zeros((col, col))
+
+    for i,j in product(range(col), range(col)):
+        D[i,j] = OP_D(i,j)
+
+    inv_IM = np.zeros_like(IM)
+
+    for i, P in enumerate(IM):
+        F = np.zeros(col)
+        for j,k in product(range(col), range(col)):
+            F[j] += D[j,k] * P[k]
+
+        inv_IM[i] = F
+
+    return inv_IM
+
+def OP_D(i,j):
+    """
+    Calculate three-point abel inversion operator Di,j
+    The formula followed Dasch 1992 (Applied Optics) which contains several typos.
+    One correction is done in function OP1 follow Martin's PhD thesis
```

Comment:
https://www.lib.utexas.edu/etd/d/2002/martinkm07836/martinkm07836.pdf

Karl Matthew Martin, UT Austin, 2002.

He mentions the typo issue on Pg 114, and gives the corrected version of some of the equations on the following pages (presumably from Dasch's 1992 errata). I've pasted an example below. Left: Martin's thesis, Right: Dasch's paper.

![image](https://cloud.githubusercontent.com/assets/6530840/12184311/7d0eb1e6-b562-11e5-8224-2016e385c783.png)